README.md

```diff
@@ -189,7 +189,14 @@ If the request parameters lead to a page outside the result set, then an empty p
 In the creation of the REST API, we've tried to follow a few specific design principles listed below
 
 ### On the Naming of Endpoints
-Names should be descriptive but reasonably short. Use nouns instead of verbs. Form compounds by concatenating words, all lower case, without punctuation.  For example: `metadatafields`, not `metadata-fields` or `MetadataFields`.
+
+We strive to align with these rules for all REST API endpoint names:
+* Endpoint names MUST use plural nouns instead of verbs.
+* When compounds are used in an endpoint name, the words MUST be concatenated, all lowercase, without punctuation. For example: `metadatafields`, NOT `metadata-fields` or `MetadataFields` or `metadataFields`.
+* Endpoint names SHOULD be descriptive, but reasonably short (<25 characters). Abbreviations are _not recommended_, unless necessary for brevity. When abbreviations are used, the REST Contract MUST spell out the meaning of the abbreviation.
+* Endpoints SHOULD be grouped with other related endpoints by category or name. 
+    * The "category" is part of the endpoint's path. For example, all configuration-related endpoints use the category "config", which means they appear under `/api/config/*`.  Similarly, all submission-related endpoints use the category "submission", which means they appear under `/api/submission/*`.
+    * Alternatively, the name of the endpoint may include a prefix to relate it to other endpoints.  For example, all workflow related endpoints start with the prefix "workflow", regardless of their category. For example: `/api/config/workflowdefinitions`, `/api/config/workflowsteps`, `/api/workflow/workflowitems`.
```

Comment:
we can add that the category is singular noun, it would be useful to describe the existing categories

Reply:
Done! I added that in the bullet just above this one:
> The "category" is the first part of the endpoint's path and is always a singular noun.